velox/functions/lib/window/NthValue.cpp

```diff
@@ -33,21 +32,38 @@ class NthValueFunction : public exec::WindowFunction {
     VELOX_CHECK_EQ(args.size(), 2);
     VELOX_CHECK_NULL(args[0].constantValue);
     valueIndex_ = args[0].index.value();
-    if (args[1].constantValue) {
+    if (args[1].type->isInteger()) {
+      VELOX_USER_CHECK(
+          args[1].constantValue,
+          "Offset must be literal for spark");
```

Comment:
Reference to Spark here is breaking the abstraction. The 'lib' is supposed to be shared code, not specialized for Spark or Presto.